File: data/src/main/java/org/oppia/data/backends/gae/model/Exploration.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

import java.util.List;
import java.util.Map;

public class Exploration {

  /*
   * Parent Hierarchy: ExplorationContainer -> Exploration
   */

  /*
   * Ignore below params
   * language_code
   */

  //TODO: ImmutableMap<> and Map<> error
  //ImmutableMap<> does not work but Map<> works
  //TODO: ImmutableList<> and List<> error

  @Json(name = "states")
  public Map<String, State> states;

  @Json(name = "param_changes")
  public List<ParamChange> paramChanges;

  @Json(name = "param_specs")
  public Map<String, ParamSpec> paramSpecs;

  @Json(name = "init_state_name")
  public String initStateName;

  @Json(name = "objective")
  public String objective;

  @Json(name = "correctness_feedback_enabled")
  public boolean correctnessFeedbackEnabled;

  @Json(name = "title")
  public String title;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/Exploration.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Exploration (

  /*
    * Ignore below params
    * language_code
    */

  @Json(name = "states") val states: Map<String, State>?,
  @Json(name = "param_changes") val paramChanges: List<ParamChange>?,
  @Json(name = "param_specs") val paramSpecs: Map<String, ParamSpec>?,
  @Json(name = "init_state_name") val initStateName: String?,
  @Json(name = "objective") val objective: String?,
  @Json(name = "correctness_feedback_enabled") val isCorrectnessFeedbackEnabled: Boolean?,
  @Json(name = "title") val title: String?

)
